cs/src/core/io/safe/OutputBuffer.cs

```diff
@@ -36,14 +36,17 @@ public virtual long Position
         }
 
         public OutputBuffer(int length = 64 * 1024)
-            : this(new byte[length])
-        {}
+            : this(buffer: null)
```

Comment:
I wanted to avoid repeating `Debug.Assert(BitConverter.IsLittleEndian)`, refactored out a private default constructor instead.

Also added a note in CHANGELOG.md